app/frontend/terrier/parts/panel-part.ts

```diff
@@ -33,12 +51,37 @@ export default abstract class PanelPart<TState> extends ContentPart<TState> {
                     header.div('.tertiary-actions', actions => {
                         this.theme.renderActions(actions, this.getActions('tertiary'))
                     })
+                    if (this.chevronSide == 'right') {
+                        this.renderChevron(header)
+                    }
                 })
             }
-            panel.div('.panel-content', ...this.contentClasses, content => {
-                this.renderContent(content)
-            })
+            if (!this.collapsed) {
+                panel.div('.panel-content', ...this.contentClasses, content => {
+                    this.renderContent(content)
+                })
+            }
+
             Fragments.panelActions(panel, this.getAllActions(), this.theme)
         })
     }
+
+    toggleCollapse() {
+        this.collapsed = !this.collapsed
+        this.dirty()
+    }
+
+    renderChevron(parent: PartTag) {
+        if (this.collapsible) {
+            parent.div('.collapsible-chevron', chev => {
+                if (this.collapsed) {
+                    this.app.theme.renderIcon(chev, `glyp-chevron_right`, 'white')
+                }
+                else {
+                    this.app.theme.renderIcon(chev, 'glyp-chevron_down', 'white')
+                }
+                chev.emitClick(this.toggleCollapseKey, {})
```

Comment:
Toggle whenever **header** is clicked, instead of just the chevron.
```suggestion
                parent.emitClick(this.toggleCollapseKey)
```